src/utils/rate-limit.js

```diff
@@ -25,27 +25,30 @@
  * @returns {function()}
  */
 export function rateLimit(win, callback, minInterval) {
-  let locker = null;
+  let locker = 0;
   let nextCallArgs = null;
 
-  const fire = args => {
-    callback.apply(null, args);
-
+  function fire(args) {
```

Comment:
I would have to use `let`s if not, because of the TDZ. This is because `fire` refers to `wait` and `wait` refers to `fire`. One of them has to come first, which would break TDZ.